### ISSUE TITLE
Warn users about pending runtime deprecations

### ISSUE THREAD

**Issue by jmcarp** [18F/cg-atlas#144]
Cloud Foundry buildpacks periodically add new language runtime versions and drop old ones: https://github.com/cloudfoundry/python-buildpack/releases/tag/v1.5.8, https://github.com/cloudfoundry/nodejs-buildpack/releases/tag/v1.5.18, etc. 

Tenants don't currently have a way of learning about changes to available runtimes until they deploy a previously working app and encounter an error. If possible, we should warn users about upcoming changes in advance. If that's not feasible, we can at least tell users where to find currently supported runtimes and offer guidance on how to keep current--a good practice there would be to avoid using the oldest supported runtime, since those are the first to be dropped.

**Comment by berndverst** [comment]
This is very interesting.

Do we have (as platform operators) have a way to run reports on the runtime a given application (or space) is using?
Maybe we can automate these notifications.

**Comment by mogul** [comment]
Yes, though it's not always accurate, particularly if they're using a custom buildpack. 

We've also discussed notifying people when there's a new buildpack version, because we don't believe it's automatically applied until they (at least) restage their application. However, I keep seeing conflicting info about that lately, with some sources implying their app _is_ automatically restaged when a new buildpack appears. We need to determine authoritatively what happens. (Pretty sure there's an existing issue/feature about this, but I'm having a hard time finding it.)

**Comment by mogul** [comment]
(Kinda buried, but [I found it](https://18f.aha.io/features/CGP-235#requirement_6284264633315295898).)

**Comment by nikzei** [comment]
PI8 75% assessment:

SHOULD: This is important for our customer experience + potentially important for prospective customers. Unsure about the size of this issue + whether we'd need to push it into PI9.

**Comment by mogul** [comment]
Definitely for PI9; reparented onto the Feature we've got slated.
